errors/registry.go

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package errors
+
+import (
+	"fmt"
+	"sync"
+)
+
+// registry represents an error type registry
+type registry struct {
+	sync.RWMutex
+	byCode map[Code]*ErrDescriptor
+}
+
+// Register registers a new error type
+func (r *registry) Register(err *ErrDescriptor) {
+	r.Lock()
+	defer r.Unlock()
+
+	if err.Code == 0 {
+		panic(fmt.Errorf("No code defined in error descriptor (message: `%s`)", err.MessageFormat))
+	}
+
+	if r.byCode[err.Code] != nil {
+		panic(fmt.Errorf("errors: Duplicate error code %v registered", err.Code))
+	}
+
+	err.registered = true
+	r.byCode[err.Code] = err
+}
+
+// Get returns the descriptor if it exists or nil otherwise
+func (r *registry) Get(code Code) *ErrDescriptor {
+	r.RLock()
+	defer r.RUnlock()
+	return r.byCode[code]
+}
+
+// reg is a global registry to be shared by packages
```

Comment:
We should probably just create code namespaces, like:
```
console: 001-1000
backend: 1001-3000
integrations: 3001-5000
```
I could put it in the api like that: every user has it's own registry with offset and codes and that automatically registers the codes in a global registry. This is how I envisaged it anyway but having it explicit would be nice indeed.